packages/common/src/hooks/useBreadcrumbs/useBreadcrumbs.tsx

```diff
@@ -0,0 +1,52 @@
+import { useState, useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { LocaleKey } from '@common/intl';
+
+type BreadcrumbNavigationAction = 'up-one' | null;
+
+export interface UrlPart {
+  path: string;
+  key: LocaleKey;
+  value: string;
+}
+
+export const useBreadcrumbs = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [urlParts, setUrlParts] = useState<UrlPart[]>([]);
+  const [navAction, setNavAction] = useState<BreadcrumbNavigationAction>(null);
+
+  useEffect(() => {
+    const parts = location.pathname.split('/');
+    const urlParts: UrlPart[] = [];
+
+    parts.reduce((fullPath, part, index) => {
+      if (part === '') return '';
+      const path = `${fullPath}/${part}`;
+
+      if (index > 1)
+        urlParts.push({
+          path,
+          key: `${part}` as unknown as LocaleKey,
+          value: part,
+        });
+      return path;
+    }, '');
+    setUrlParts(urlParts);
+  }, [location]);
+
+  useEffect(() => {
+    if (!navAction) return;
+    if (navAction === 'up-one') navigateUpOne();
+    setNavAction(null);
+  }, [navAction]);
+
+  const navigateUpOne = () => {
+    console.log('Up one breadcrumb');
+    console.log('parts', urlParts);
+    if (urlParts.length < 2) return;
+    navigate(urlParts[urlParts.length - 2]?.path as string);
+  };
```

Comment:
Initially tried to return the "navigateUpOne" function directly without having `useEffect`. Problem was the kbar Action would get a previous "urlParts" state (presumably a stale closure). Open to other suggestions?

Reply:
Oooh -- I didn't realise you could pass a dependency array to it! How did you know that? -- I didn't see any mention on the [kbar docs](https://kbar.vercel.app/docs/concepts/actions) (which are frustratingly sparse tbh)

Will definitely update this then, that'll make things simpler. Thanks

Reply:
Ah yea. I knew the docs weren't great so I just cheated and went straight to the code https://github.com/timc1/kbar/blob/8f75115062504876e1a7992aad5a3c0ce549f7fa/src/useRegisterActions.tsx#L12